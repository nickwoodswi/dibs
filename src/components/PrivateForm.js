import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { Route } from 'react-router-dom'
import StartDateSelector from './StartDateSelector'
import EndDateSelector from './EndDateSelector'
import MultiHoldSelector from './MultiHoldSelector'
import SingleHold from './SingleHold'
import './PrivateForm.css'

class PrivateForm extends Component {
    render() {
        return(
        <div className="private-form">
            <div className="hold-type-header">PLACING HOLD(S) ON:</div>
            <div className="hold-type-buttons">
                <Link to="/promoter/single"><button id="single" name="hold_type" value="single" onClick={e => this.props.handleState('hold_type', e.target.value)}>A SINGLE DATE</button></Link>
                <Link to="/promoter/range"><button id="range" name="hold_type" value="range" onClick={e => this.props.handleState('hold_type', e.target.value)}>A RANGE OF DATES</button></Link>
                <Link to="/promoter/multi"><button id="multiple" name="hold_type" value="multiple" onClick={e => this.props.handleState('hold_type', e.target.value)}>MULTIPLE DATES</button></Link>
            </div>
            <Route path="/private/single">
                <SingleHold 
                    appState={this.props.appState} 
                    handleSingleHold={this.props.handleSingleHold}
                    submitForm={this.props.submitForm} />
            </Route>
            <Route path="/private/range">
                <div className="range-date-selector">
                    <StartDateSelector 
                        appState={this.props.appState} 
                        handleState={this.props.handleState} />
                    <EndDateSelector 
                        appState={this.props.appState} 
                        handleState={this.props.handleState} />
                </div>
        </Route>
        <Route path="/private/multi">
            <MultiHoldSelector 
                appState={this.props.appState} 
                handleState={this.props.handleState} 
                addHold={this.props.addHold}
                deleteHold={this.props.deleteHold}
                submitForm={this.props.submitForm} />
        </Route>
        <div className="performance-type-detail">
            FOR:
            <input type="text" onChange={e => this.props.handleState('contact_act', e.target.value)} value={this.props.appState.contact_act} placeholder="Short Event Description" />
        </div>
        
        <div className="org-detail">
            Your Organization: <input type="text" onChange={e => this.props.handleState('contact_org', e.target.value)} value={this.props.appState.contact_org} placeholder="Name, if applicable" />
            Phone: <input type="text" onChange={e => this.props.handleState('contact_phone', e.target.value)} value={this.props.appState.contact_phone} placeholder="###" />
            Mailing Address:
            <input type="text" onChange={e => this.props.handleState('org_street1', e.target.value)} value={this.props.appState.org_street1} placeholder="Street 1" />
            <input type="text" onChange={e => this.props.handleState('org_street2', e.target.value)} value={this.props.appState.org_street2} placeholder="Street 2" />
            <input type="text" onChange={e => this.props.handleState('org_city', e.target.value)} value={this.props.appState.org_city} placeholder="City" />
            <input type="text" onChange={e => this.props.handleState('org_country', e.target.value)} value={this.props.appState.org_country} placeholder="Country" />
            <input type="text" onChange={e => this.props.handleState('org_zip', e.target.value)} value={this.props.appState.org_zip} placeholder="Postal Code" />
        </div>

            <div className="event-detail">
                Provide any specific event needs, ideas, or details here:
                <input type="text" id="event-detail" onChange={e => this.props.handleState('event_detail', e.target.value)} value={this.props.appState.event_detail} />
            </div>

            <div className="place-hold-button-container">
                <Link to="/holdconfirmation"><button onClick={e => this.props.submitForm()}>PLACE HOLDS</button></Link>
            </div>
        
        </div>
        )
    }
}

export default PrivateForm