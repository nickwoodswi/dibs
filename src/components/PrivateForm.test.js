import React from 'react';
import { render } from '@testing-library/react';
import PrivateForm from './AgentForm';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders private form', () => {
    render(<MemoryRouter><PrivateForm /></MemoryRouter>);
});

it('PrivateForm renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><PrivateForm /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});