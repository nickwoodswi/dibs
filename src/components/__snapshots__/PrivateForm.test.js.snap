// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateForm renders the UI as expected 1`] = `
<div
  className="agent-form"
>
  <div
    className="hold-type-header"
  >
    PLACING HOLD(S) ON:
  </div>
  <div
    className="hold-type-buttons"
  >
    <a
      href="/agent/single"
      onClick={[Function]}
    >
      <button
        id="single"
        name="hold_type"
        onClick={[Function]}
        value="single"
      >
        A SINGLE DATE
      </button>
    </a>
    <a
      href="/agent/range"
      onClick={[Function]}
    >
      <button
        id="range"
        name="hold_type"
        onClick={[Function]}
        value="range"
      >
        A RANGE OF DATES
      </button>
    </a>
    <a
      href="/agent/multi"
      onClick={[Function]}
    >
      <button
        id="multiple"
        name="hold_type"
        onClick={[Function]}
        value="multiple"
      >
        MULTIPLE DATES
      </button>
    </a>
  </div>
  <div
    className="performance-type-detail"
  >
    FOR:
    <input
      onChange={[Function]}
      placeholder="Band, Artist, or Performer Title/Name"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Your Home ZIP Code"
      type="text"
    />
    <input
      id="band"
      name="performance_type"
      onChange={[Function]}
      type="radio"
      value="band"
    />
    Band
    <input
      id="solo"
      name="performance_type"
      onChange={[Function]}
      type="radio"
      value="solo"
    />
    Solo Performer
    <input
      id="dj"
      name="performance_type"
      onChange={[Function]}
      type="radio"
      value="dj"
    />
    DJ
  </div>
  <div
    className="media"
  >
    Please link to your client's music, or a look at their performance:
    <input
      onChange={[Function]}
      placeholder="Spotify Artist ID"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="YouTube Channel or Video"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Facebook Page"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Bandcamp"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Website"
      type="text"
    />
  </div>
  <div
    className="support-need"
  >
    Do you need us to book local support?
    <input
      id="support_need"
      name="support_need"
      onChange={[Function]}
      type="radio"
      value="yes"
    />
    Yes
    <input
      id="support_need"
      name="support_need"
      onChange={[Function]}
      type="radio"
      value="no"
    />
    No
  </div>
  <div
    className="package-support-header"
  >
    PACKAGE SUPPORT
  </div>
  <div
    className="performance-type-detail"
  >
    <input
      onChange={[Function]}
      placeholder="Direct Support"
      type="text"
    />
    <input
      id="band-2"
      name="support_type-2"
      onChange={[Function]}
      type="radio"
      value="band"
    />
    Band
    <input
      id="solo-2"
      name="support_type-2"
      onChange={[Function]}
      type="radio"
      value="solo"
    />
    Solo Performer
    <input
      id="dj-2"
      name="support_type-2"
      onChange={[Function]}
      type="radio"
      value="dj"
    />
    DJ
  </div>
  <div
    className="media"
  >
    <input
      onChange={[Function]}
      placeholder="Spotify Artist ID"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="YouTube Video or Channel"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Facebook Page"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Bandcamp"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Website"
      type="text"
    />
  </div>
  <div
    className="performance-type-detail"
  >
    <input
      onChange={[Function]}
      placeholder="Opener"
      type="text"
    />
    <input
      id="band-3"
      name="support_type3"
      onChange={[Function]}
      type="radio"
      value="band"
    />
    Band
    <input
      id="solo-3"
      name="support_type3"
      onChange={[Function]}
      type="radio"
      value="solo"
    />
    Solo Performer
    <input
      id="dj-3"
      name="support_type3"
      onChange={[Function]}
      type="radio"
      value="dj"
    />
    DJ
  </div>
  <div
    className="media"
  >
    <input
      onChange={[Function]}
      placeholder="Spotify Artist ID"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Youtube Video or Channel"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Facebook Page"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Bandcamp"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Website"
      type="text"
    />
    <br />
  </div>
  <div
    className="performance-type-detail"
  >
    <input
      onChange={[Function]}
      placeholder="Second Opener Title"
      type="text"
    />
    <input
      id="band-4"
      name="support_type4"
      onChange={[Function]}
      type="radio"
      value="band"
    />
    Band
    <input
      id="solo-4"
      name="support_type4"
      onChange={[Function]}
      type="radio"
      value="solo"
    />
    Solo Performer
    <input
      id="dj-4"
      name="support_type4"
      onChange={[Function]}
      type="radio"
      value="dj"
    />
    DJ
  </div>
  <div
    className="media"
  >
    <input
      onChange={[Function]}
      placeholder="Spotify Artist ID"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="YouTube Video or Channel"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Facebook Page"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Bandcamp"
      type="text"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="Website"
      type="text"
    />
    <br />
  </div>
  <div
    className="event-detail"
  >
    Provide any specific technical needs or details here:
    <input
      id="event-detail"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="org-detail"
  >
    Agency: 
    <input
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
    Phone: 
    <input
      onChange={[Function]}
      placeholder="###"
      type="text"
    />
    <br />
    <br />
    Mailing Address:
    <input
      onChange={[Function]}
      placeholder="Street 1"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Street 2"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="City"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Country"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Postal Code"
      type="text"
    />
  </div>
  <div
    className="place-hold-button-container"
  >
    <a
      href="/holdconfirmation"
      onClick={[Function]}
    >
      <button
        onClick={[Function]}
      >
        PLACE HOLDS
      </button>
    </a>
  </div>
</div>
`;
